src/App.tsx

```diff
@@ -1,27 +1,84 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [hasError, setHasError] = useState(false);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleAll = useCallback(async () => {
+    setHasError(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+    try {
+      const allGoods = await getAll();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setGoods(allGoods);
+    } catch {
+      setHasError(true);
+    }
+  }, []);
```

Comment:
good job, but i think you dont need this empty array in the end, try to delete it, does it work without it?

Reply:
If the second argument is omitted, the value will never be memoized, and the useCallback and the useMemo doesn't do anything.
@vladyslav78292 bad idea)